components/shopify/shopify.app.mjs

```diff
@@ -208,50 +221,227 @@ module.exports = {
+        \`archived\`: The product is no longer being sold and isn't available to customers on sales channels and apps.
+        \`draft\`: The product isn't ready to sell and is unavailable to customers on sales channels and apps. By default, duplicated and unarchived products are set to draft
+      `),
+      optional: true,
+      options: [
+        "active",
+        "archived",
+        "draft",
+      ],
+    },
+    images: {
       type: "string[]",
-      label: "Fields",
-      description: `A comma-separated list of fields to include in the response
-        Check out [Shopify Product API](https://shopify.dev/api/admin-rest/2022-01/resources/product#resource_object) for options`,
+      label: "Images",
+      description: toSingleLineString(`
+        A list of product base64 encoded image objects.
+        Each one represents an image associated with the product or a link that will be downloaded by Shopify.
+        Example: \`["R0lGODlhAQABAIAAAAAAAAAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==","http://example.com/rails_logo.gif"]\`.
+        More information at [Shopify Product API](https://shopify.dev/api/admin-rest/2022-01/resources/product#[post]/admin/api/2022-01/products.json)
+      `),
+      optional: true,
+    },
+    options: {
+      type: "string[]",
+      label: "Options",
+      description: toSingleLineString(`
+        The custom product properties.
+        For example, Size, Color, and Material. Each product can have up to 3 options and each option value can be up to 255 characters.
+        Product variants are made of up combinations of option values. Options cannot be created without values.
+        To create new options, a variant with an associated option value also needs to be created.
+        Example: \`[{"name":"Color","values":["Blue","Black"]},{"name":"Size","values":["155","159"]}]\`
+      `),
+      optional: true,
+    },
+    variants: {
+      type: "string[]",
+      label: "Product Variants",
+      description: toSingleLineString(`
+        An array of product variants, each representing a different version of the product.
+        The position property is read-only. The position of variants is indicated by the order in which they are listed.
+        Example: \`[{"option1":"First","price":"10.00","sku":"123"},{"option1":"Second","price":"20.00","sku":"123"}]\`
+      `),
+      optional: true,
+    },
+    tags: {
+      type: "string[]",
+      label: "Tags",
+      description: toSingleLineString(`
+        A string of comma-separated tags that are used for filtering and search. A product can have up to 250 tags. Each tag can have up to 255 characters.
+        Example: \`"Barnes & Noble","Big Air","John's Fav"\`
+      `),
+      optional: true,
+    },
+    option: {
+      type: "string",
+      label: "Title",
+      description: "The option name of the product variant",
+    },
+    imageId: {
```

Comment:
Using the legacy  workflow builder, I was able to add async options by redefining the `imageId` prop rather than inheriting from `shopify.app.mjs`.

```js
      props.imageId = {
        type: "string",
        label: "Image ID",
        description: "The unique numeric identifier for a product's image. The image must be associated to the same product as the variant",
        optional: true,
        options: async ({ prevContext }) => {
          const productId = this.productId;
          const { nextPageParameters } = prevContext;
          return this.shopify.listImagesOptions(productId, nextPageParameters);
        },
      };
```

But there seems to be an issue where the async prop value is reset whenever another prop is changed. And this approach doesn't seem to work using the new workflow builder. So I think we should omit async options from the `imageId` prop in **Search Product Variant** for now.